Neos.Cache/Classes/Backend/RedisBackend.php

```diff
@@ -535,7 +543,8 @@ private function getRedisClient(): \Redis
             }
         }
 
-        if ($this->password !== '' && !$redis->auth($this->password)) {
+        $credentials = array_values(array_filter([$this->username, $this->password]));
```

Comment:
I'd suggest to make this code a bit more readable – as not everyone is familiar `with array_filter()` and the fact that `$redis->auth()` accepts both a string or an array (see [docs](https://github.com/phpredis/phpredis/?tab=readme-ov-file#auth)).

Also, I'd prefer strict comparisons over weak ones like `empty()'.

Howe about something like this?

```php
        if ($this->username !== '' && $this->password !== '') {
            $result = $redis->auth($this->password);
            if ($result === false) {
                throw new CacheException(sprintf('Redis authentication failed, using username "%s" and a %s bytes long password.', $this->username, strlen($this->password)), 1725607160);
            }
        } elseif ($this->password !== '') {
            $result = $redis->auth($this->password);
            if ($result === false) {
                throw new CacheException(sprintf('Redis authentication failed, using a %s bytes long password.', strlen($this->password)), 1502366200);
            }
        } else {
            $result = $redis->auth();
            if ($result === false) {
                throw new CacheException('Redis authentication with empty password failed.', 1725607324);
            }
        }
```